datatable/java/datatable/src/test/java/io/cucumber/datatable/DataTableTypeRegistryTest.java

```diff
@@ -37,8 +46,132 @@ public Place transform(DataTable table) {
         }));
     }
 
+    @Test
+    public void returns_null_data_table_type_if_none_match_and_no_default_registered() {
+
+        registry.defineDataTableType(DataTableType.cell(Place.class));
+        registry.defineDataTableType(DataTableType.entry(Place.class));
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructType(Place.class));
+
+        assertNull(lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_null_data_table_type_for_cell_if_no_default_registered() throws Exception {
+
+        registerDefaultEntryTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForCell());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_null_data_table_type_for_entry_if_no_default_registered() throws Exception {
+
+        registerDefaultCellTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForEntry());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+
+
+    @SuppressWarnings({"unchecked", "ConstantConditions"})
+    @Test
+    public void returns_default_data_table_type_for_cell_if_none_match_and_default_registered() {
+
+        registerCellEntryAndDefaults();
+
+        String here="here";
+        DataTableType lookupTableTypeByTypeForCell = registry.lookupTableTypeByType(constructTypeForCell());
+        List<List<Place>> transformedCells = (List<List<Place>>) lookupTableTypeByTypeForCell.transform(singletonList(singletonList(here)));
+
+        assertEquals(1,transformedCells.size());
+        assertEquals(1,transformedCells.get(0).size());
+        assertEquals(here,transformedCells.get(0).get(0).name);
```

Comment:
This can be replaced with a single assert against a list of lists of places. It is okay to make place implement equals and hashcode.